src/pni/nexus/path/path.cpp

```diff
@@ -38,6 +38,38 @@ Path::Path():
             _elements()
 {}
 
+Path::Path(const std::string &str):
+  _file_name(),
+  _attribute_name(),
+  _elements()
+{
+  *this = Path::from_string(str);
+  if(size()!=0)
+  {
+    auto new_pos = std::remove_if(_elements.begin(),_elements.end(),
+                                  [](const Element &element)
+                                  { return element.first == "."; });
+    if(new_pos!=_elements.end())
+      _elements.erase(new_pos);
+  }
```

Comment:
This code is the same in the constructors, maybe it could be replaced by a method